src/tape/ensemble.py

```diff
@@ -607,41 +607,68 @@ def calc_nobs(self, by_band=False, label="nobs", temporary=True):
         """
 
         if by_band:
-            band_counts = (
-                self._source.groupby([self._id_col])[self._band_col]  # group by each object
-                .value_counts()  # count occurence of each band
-                .to_frame()  # convert series to dataframe
-                .rename(columns={self._band_col: "counts"})  # rename column
-                .reset_index()  # break up the multiindex
-                .categorize(columns=[self._band_col])  # retype the band labels as categories
-                .pivot_table(values=self._band_col, index=self._id_col, columns=self._band_col, aggfunc="sum")
-            )  # the pivot_table call makes each band_count a column of the id_col row
-
             # repartition the result to align with object
             if self._object.known_divisions:
-                band_counts.divisions = self._source.divisions
-                band_counts = band_counts.repartition(divisions=self._object.divisions)
+                # Grab these up front to help out the task graph
+                id_col = self._id_col
+                band_col = self._band_col
+
+                # Get the band metadata
+                unq_bands = np.unique(self._source[band_col])
```

Comment:
Nit: I'm not fully certain about the performance differences, but I feel `self._source[band_col].unique()` might be preferable and slightly more readable